src/main/java/com/moodmate/moodmatebe/domain/chat/application/ChatRoomService.java

```diff
@@ -0,0 +1,29 @@
+package com.moodmate.moodmatebe.domain.chat.application;
+
+import com.moodmate.moodmatebe.domain.chat.exception.ChatRoomNotFoundException;
+import com.moodmate.moodmatebe.domain.chat.redis.RedisSubscriber;
+import com.moodmate.moodmatebe.domain.chat.redis.exception.ConnectionException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.redis.listener.ChannelTopic;
+import org.springframework.data.redis.listener.RedisMessageListenerContainer;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ChatRoomService {
+    private final RedisMessageListenerContainer redisMessageListener;
+    private final RedisSubscriber redisSubscriber;
+
+    public void enterChatRoom(Long roomId) throws ChatRoomNotFoundException, ConnectionException {
+        ChannelTopic topic = new ChannelTopic("/sub/chat/" + roomId);
+        try{
+            redisMessageListener.addMessageListener(redisSubscriber, topic);
+        }catch (Exception e){
+            if(e.getCause() instanceof ChatRoomNotFoundException){
+                throw new ChatRoomNotFoundException();
+            }else if (e.getCause() instanceof ConnectionException){
+                throw new ConnectionException();
```

Comment:
실무자분들께 물어봤는데, 일반적으로 예외처리는 아래 방식대로 한다고 해서 아래 방식을 추천드릴게요!
```suggestion
    try {
        redisMessageListener.addMessageListener(redisSubscriber, topic);
    } catch (ChatRoomNotFoundException e) {
        throw e; // 원래 예외를 재활용
    } catch (ConnectionException e) {
        throw e; // 원래 예외를 재활용
    } catch (Exception e) {
        // 다른 일반 예외 처리
    }
```

Reply:
감사합니다! 수정했습니다 :)

Reply:
헉 대박!! 완전 꿀팁 감사합니다!👍🏻